src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -108,78 +102,93 @@
+        private static Map<String, Class<? extends RandomGenerator>> createFactoryMap() {
+            return Map.ofEntries(
+                    entry(java.security.SecureRandom.class),
+                    entry(java.util.Random.class),
+                    entry(java.util.SplittableRandom.class),
+                    entry(jdk.internal.random.L32X64MixRandom.class),
+                    entry(jdk.internal.random.L64X128MixRandom.class),
+                    entry(jdk.internal.random.L64X128StarStarRandom.class),
+                    entry(jdk.internal.random.L64X256MixRandom.class),
+                    entry(jdk.internal.random.L64X1024MixRandom.class),
+                    entry(jdk.internal.random.L128X128MixRandom.class),
+                    entry(jdk.internal.random.L128X256MixRandom.class),
+                    entry(jdk.internal.random.L128X1024MixRandom.class),
+                    entry(jdk.internal.random.Xoroshiro128PlusPlus.class),
+                    entry(jdk.internal.random.Xoshiro256PlusPlus.class)
+            );
+        }
+
+        private static SimpleImmutableEntry<String, Class<? extends RandomGenerator>>
+        entry(Class<? extends RandomGenerator> rgClass) {
+            return new SimpleImmutableEntry<>(rgClass.getSimpleName(), rgClass);
         }
     }
 
     /**
      * Private constructor.
      *
-     * @param provider  Provider class to wrap.
+     * @param rgClass  Random generator class.
      */
-    private RandomGeneratorFactory(Provider<? extends RandomGenerator> provider) {
-        this.provider = provider;
+    private RandomGeneratorFactory(Class<? extends RandomGenerator> rgClass) {
+        this.rgClass = rgClass;
     }
 
     /**
      * Returns the factory map, lazily constructing map on first call.
      *
-     * @return Map of RandomGeneratorFactory classes.
+     * @return Map of random generator classes.
      */
-    private static Map<String, Provider<? extends RandomGenerator>> getFactoryMap() {
+    private static Map<String, Class<? extends RandomGenerator>> getFactoryMap() {
         return FactoryMapHolder.FACTORY_MAP;
     }
 
     /**
-     * Return the annotation for the specified provider.
+     * Return the annotation for the specified random generator.
      *
-     * @return RandomGeneratorProperties annotation for the specified provider.
+     * @return RandomGeneratorProperties annotation for the specified random generator.
      */
      private RandomGeneratorProperties getProperties() {
-        if (properties == null) {
-            synchronized (provider) {
-                if (properties == null) {
-                    properties = provider.type().getDeclaredAnnotation(RandomGeneratorProperties.class);
-                    Objects.requireNonNull(properties, provider.type() + " missing annotation");
+        if (properties == null) {  // volatile load
+            synchronized (rgClass) {
```

Comment:
The synchronization on `rgClass` to intialize an instance field `properties` appears odd here. I think this should be synchronized on `this`.